packages/@stylexjs/babel-plugin/src/index.js

```diff
@@ -389,6 +389,137 @@ function getLogicalFloatVars(rules: Array<Rule>): string {
     : '';
 }
 
+function optimizeDefineVarsRules(rules: Array<Rule>): {
+  optimizedRules: Array<Rule>,
+  varsToInline: Map<string, string>,
+} {
+  const defineVarsVariables: Map<string, string> = new Map();
+  const varUsageCount: Map<string, number> = new Map();
+
+  for (const [, ruleObj] of rules) {
+    const { ltr, rtl } = ruleObj;
+    const isDefineVarsRule = /^(?:@[^{]*\{)?:root,\s*\.[\w-]+\{/.test(ltr);
```

Comment:
We probably want to change the metadata generated by `defineVars` before this PR and avoid regex-based fragile implementations.